api/infrastructure/parallelcluster-api.yaml

```diff
@@ -0,0 +1,246 @@
+            Location: !Ref ApiDefinitionS3Uri
+      Domain:
+        DomainName: !Ref CustomDomainName
+        CertificateArn: !Ref CustomDomainCertificate
+        Route53:
+          HostedZoneId: !Ref CustomDomainHostedZoneId
+
+  ApiGatewayApiWithoutCustomDomain:
+    Condition: DoesNotUseCustomDomain
+    Type: AWS::Serverless::Api
+    Properties:
+      Tags:
+        parallelcluster: !Ref ParallelClusterVersion
+      StageName: prod
+      DefinitionBody:
+        Fn::Transform:
+          Name: AWS::Include
+          Parameters:
+            Location: !Ref ApiDefinitionS3Uri
+
+  APIGatewayExecutionRole:
+   Type: AWS::IAM::Role
+   Properties:
+     AssumeRolePolicyDocument:
+       Version: 2012-10-17
+       Statement:
+         - Effect: Allow
+           Principal:
+             Service:
+               - apigateway.amazonaws.com
+           Action:
+             - 'sts:AssumeRole'
+     Policies:
+       - PolicyName: lambda-invoke
+         PolicyDocument:
+           Version: '2012-10-17'
+           Statement:
+             - Effect: Allow
+               Action: lambda:InvokeFunction
+               Resource: !GetAtt ParallelClusterFunction.Arn
+
+  ParallelClusterFunction:
+    Type: AWS::Serverless::Function
+    DependsOn: LambdaLogGroup
+    Properties:
+      PackageType: Image
+      Tracing: Active
+      Tags:
+        parallelcluster: !Ref ParallelClusterVersion
+      Policies:
+        - AWSLambdaBasicExecutionRole
+        - AWSXRayDaemonWriteAccess
+        - AdministratorAccess
+      ImageUri: !If [UsesCustomEcrImageUri, !Ref EcrImageUri, !Ref PublicEcrImageUri]
+
+  LambdaLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      RetentionInDays: 30
+
+  InstanceRole:
+    Condition: DoesNotUseCustomEcrImageUri
+    Type: AWS::IAM::Role
+    Properties:
+      ManagedPolicyArns:
+        - Fn::Sub: arn:${AWS::Partition}:iam::aws:policy/AmazonSSMManagedInstanceCore
+        - Fn::Sub: arn:${AWS::Partition}:iam::aws:policy/EC2InstanceProfileForImageBuilderECRContainerBuilds
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action:
+              - sts:AssumeRole
+            Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+        Version: '2012-10-17'
+      Path: /executionServiceEC2Role/
+
+  InstanceProfile:
+    Condition: DoesNotUseCustomEcrImageUri
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Path: /executionServiceEC2Role/
+      Roles:
+        - !Ref InstanceRole
+
+  InfrastructureConfiguration:
+    Condition: DoesNotUseCustomEcrImageUri
+    Type: AWS::ImageBuilder::InfrastructureConfiguration
+    Properties:
+      Name: ParallelClusterImageBuilderInfrastructureConfiguration
+      InstanceProfileName: !Ref InstanceProfile
+      TerminateInstanceOnFailure: true
+
+  PrivateEcrRepository:
+    Condition: DoesNotUseCustomEcrImageUri
+    Type: AWS::ECR::Repository
+    Properties:
+      RepositoryName: !Sub
```

Comment:
I know I was the one suggesting this in the first place but probably better to remove this name and just tag the repo so to avoid conflicts if creating multiple API stacks in the same region

Reply:
In light of our discussion regarding the names of `AWS::ImageBuilder::InfrastructureConfiguration` and `AWS::ImageBuilder::ContainerRecipe`, what about following the same pattern and putting `aws-parallelcluster-${Version}-${StackIdSuffix}`?